kivy/uix/videoplayer.py

```diff
@@ -501,47 +499,58 @@ def _trigger_video_load(self, *largs):
                                                            -1)
         ev()
 
+    def _try_load_default_thumbnail(self, *largs):
+        if not self.thumbnail:
+            filename = self.source.rsplit('.', 1)
+            thumbnail = filename[0] + '.png'
+            if exists(thumbnail):
+                self._load_thumbnail(thumbnail)
+
+    def _try_load_default_annotations(self, *largs):
+        if not self.annotations:
+            filename = self.source.rsplit('.', 1)
+            annotations = filename[0] + '.jsa'
+            if exists(annotations):
+                self._load_annotations(annotations)
+
     def on_source(self, instance, value):
-        # we got a value, try to see if we have an image for it
-        self._load_thumbnail()
-        self._load_annotations()
+        # By default, VideoPlayer should look for thumbnail and annotations
+        # with the same filename (except extension) of the source (video) file.
+        Clock.schedule_once(self._try_load_default_thumbnail, -1)
+        Clock.schedule_once(self._try_load_default_annotations, -1)
         if self._video is not None:
             self._video.unload()
             self._video = None
         if value:
             self._trigger_video_load()
 
+    def on_thumbnail(self, instance, value):
+        if value:
+            self._load_thumbnail(value)
```

Comment:
Do we need to reset to default if value is None? Or maybe clear it?

Reply:
Now the user can clear it. Added a test that checks it.